engine/gamesys/src/gamesys/components/comp_gui.cpp

```diff
@@ -3293,6 +3318,51 @@ namespace dmGameSystem
         return dmGameObject::RESULT_OK;
     }
 
+
+    dmGameObject::Result CompGuiRegisterSetPropertyFn(dmhash_t property_hash, CompGuiPropertySetterFn setter)
+    {
+        if (g_CompGuiPropertySetters.Full())
+        {
+            uint32_t capacity = g_CompGuiPropertySetters.Capacity();
+            capacity = capacity > 0 ? capacity * 2 : 16;
+            g_CompGuiPropertySetters.SetCapacity(capacity * 2, capacity);
+        }
+        g_CompGuiPropertySetters.Put(property_hash, setter);
+        return dmGameObject::RESULT_OK;
+    }
+
+    dmGameObject::Result CompGuiRegisterGetPropertyFn(dmhash_t property_hash, CompGuiPropertyGetterFn getter)
+    {
+        if (g_CompGuiPropertyGetters.Full())
+        {
+            uint32_t capacity = g_CompGuiPropertyGetters.Capacity();
+            capacity = capacity > 0 ? capacity * 2 : 16;
+            g_CompGuiPropertyGetters.SetCapacity(capacity * 2, capacity);
+        }
+        g_CompGuiPropertyGetters.Put(property_hash, getter);
+        return dmGameObject::RESULT_OK;
+    }
+
+    dmGameObject::Result CompGuiUnregisterSetPropertyFn(dmhash_t property_hash)
+    {
+        if (g_CompGuiPropertySetters.Get(property_hash) != 0)
+        {
+            g_CompGuiPropertySetters.Erase(property_hash);
+            return dmGameObject::RESULT_OK;
+        }
+        return dmGameObject::RESULT_PROPERTY_ALREADY_SET;
```

Comment:
I feel the name is a bit off. It's not a property, but a function.
Also, I feel that perhaps our enum list is a bit random at times.
Perhaps there is an enum name that fits multiple cases. E.g. RESULT_ALREADY_REGISTERED, RESULT_ALREADY_EXISTS, or something.